### ISSUE TITLE
[Paper Note] OctoThinker: Mid-training Incentivizes Reinforcement Learning Scaling, Zengzhi Wang+, arXiv'25

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#2107]
# URL
- https://arxiv.org/abs/2506.20512v1
# Authors
  - Zengzhi Wang
  - Fan Zhou
  - Xuefeng Li
  - Pengfei Liu
# Abstract
  - Different base language model families, such as Llama and Qwen, exhibit divergent behaviors during post-training with reinforcement learning (RL), especially on reasoning-intensive tasks. What makes a base language model suitable for reinforcement learning? Gaining deeper insight into this question is essential for developing RL-scalable foundation models of the next generation. In this work, we investigate how mid-training strategies shape RL dynamics, focusing on two representative model families: Qwen and Llama. Our study reveals that (1) high-quality mathematical corpora, such as MegaMath-Web-Pro, significantly improve both base model and RL performance, while existing alternatives (e.g., FineMath-4plus) fail to do so; (2) further adding QA-style data, particularly long chain-of-thought (CoT) reasoning examples, enhances RL outcomes, and instruction data further unlocks this effect; (3) while long-CoT improves reasoning depth, it can also induce verbosity of model responses and unstability of RL training, underscoring the importance of data formatting; (4) scaling mid-training consistently leads to stronger downstream RL performance. Building on these insights, we introduce a two-stage mid-training strategy, Stable-then-Decay, in which base models are first trained on 200B tokens with a constant learning rate, followed by 20B tokens across three CoT-focused branches with learning rate decay. This yields OctoThinker, a family of models demonstrating strong RL compatibility and closing the performance gap with more RL-friendly model families, i.e., Qwen. We hope our work will help shape pre-training strategies for foundation models in the RL era. To support further research, we release our open-source models along with a curated math reasoning-intensive corpus of over 70 billion tokens (i.e., MegaMath-Web-Pro-Max).
# Translation (by gpt-4o-mini)
- 異なるベース言語モデルファミリー、例えばLlamaやQwenは、特に推論を重視したタスクにおいて、強化学習（RL）によるポストトレーニング中に異なる挙動を示します。どのような要素がベース言語モデルを強化学習に適したものにするのでしょうか？この問いに対する深い洞察を得ることは、次世代のRLスケーラブルな基盤モデルを開発するために不可欠です。本研究では、中間トレーニング戦略がRLのダイナミクスにどのように影響を与えるかを調査し、代表的なモデルファミリーであるQwenとLlamaに焦点を当てます。我々の研究は、以下のことを明らかにしました。(1) MegaMath-Web-Proのような高品質の数学コーパスは、ベースモデルとRLのパフォーマンスの両方を大幅に向上させる一方で、既存の代替手段（例：FineMath-4plus）はそうではない。(2) QAスタイルのデータ、特に長い連鎖的思考（CoT）推論の例を追加することで、RLの結果が向上し、指示データがこの効果をさらに引き出す。(3) 長いCoTは推論の深さを改善する一方で、モデルの応答の冗長性やRLトレーニングの不安定性を引き起こす可能性があり、データフォーマットの重要性を強調しています。(4) 中間トレーニングのスケーリングは、一貫して強力な下流RLパフォーマンスをもたらします。これらの洞察に基づき、我々はStable-then-Decayという二段階の中間トレーニング戦略を導入します。この戦略では、ベースモデルが最初に200Bトークンで一定の学習率でトレーニングされ、その後、学習率を減少させながら3つのCoTに焦点を当てたブランチで20Bトークンが使用されます。これにより、OctoThinkerというモデルファミリーが生まれ、強力なRL適合性を示し、QwenのようなよりRLフレンドリーなモデルファミリーとのパフォーマンスギャップを縮めます。我々の研究がRL時代の基盤モデルの事前トレーニング戦略を形成する手助けとなることを期待しています。さらなる研究を支援するために、我々はオープンソースのモデルと70億トークン以上の数学推論集約コーパス（すなわち、MegaMath-Web-Pro-Max）を公開します。
# Summary (by gpt-4o-mini)
- 異なるベース言語モデル（LlamaやQwen）の強化学習（RL）における挙動を調査し、中間トレーニング戦略がRLのダイナミクスに与える影響を明らかに。高品質の数学コーパスがモデルのパフォーマンスを向上させ、長い連鎖的思考（CoT）がRL結果を改善する一方で、冗長性や不安定性を引き起こす可能性があることを示す。二段階の中間トレーニング戦略「Stable-then-Decay」を導入し、OctoThinkerモデルファミリーを開発。オープンソースのモデルと数学推論コーパスを公開し、RL時代の基盤モデルの研究を支援することを目指す。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/sinclairwang1/status/1938244843857449431?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
mid-trainingの観点から、post trainingにおけるRLがスケーリングする条件をsystematicallyに調査している模様

**Comment by AkihikoWatanabe** [comment]
論文中にはmid-training[^1]の定義が記述されている:

<img width="808" height="353" alt="Image" src="https://github.com/user-attachments/assets/da206d3d-f811-4d69-8210-a1d0816c827f" />

[^1]: mid-trainingについてはコミュニティの間で厳密な定義はまだ無くバズワードっぽく使われている、という印象を管理人は抱いており、本稿は文献中でmid-trainingを定義する初めての試みという所感

**Comment by AkihikoWatanabe** [comment]
ざっくり言うと、上記論文中の定義でいうと、mid-trainingとは特定の目的を実現するための事前学習と事後学習の間における中間に位置する学習ステージであり、たとえば特定ドメインの知識の獲得や、事後学習の準備などが含まれる。

最近だと、コーディングや論理的思考力などの特定ドメインの能力の獲得や、context長を伸ばすために実施したり、特定ドメインでのRLの前段階としてSFTを通じてwarm-upをする（reasoning能力の獲得等）などが中間学習で実施されている印象がある。
